### PR TITLE
Asset serialization

### PR DESCRIPTION
**Pull request by Boaph5Du**
+ Asset serialization for token-tx signing
+ Automated stylefixes in package.json

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -12,12 +12,18 @@ export const intToVlq = (num) => {
...
+export const outputBytes = (out, tokenIds) => {
   let res = intToVlq(out.value);
   res = Buffer.concat([res, Buffer.from(out.ergoTree, 'hex')]);
   res = Buffer.concat([res, intToVlq(out.creationHeight)]);
 
   res = Buffer.concat([res, intToVlq(out.assets.length)]);
+  for (let i = 0; i < out.assets.length; i += 1) {
+    const t = out.assets[i].tokenId;
+    const n = tokenIds.indexOf(t);
+    res = Buffer.concat([res, intToVlq(n)]);
```
Comment (deadit) [inline src/utils.js]: What will happen if `n` be `-1`? This moment thought out?

File: src/utils.js [hunk 2]
```diff
@@ -40,7 +46,20 @@ export const inputBytes = (i) => {
   return res;
 };
 
-export const distinctTokenList = () => []; // TODO: rework this
+export const distinctTokenList = (outputs) => {
+  const tokenList = outputs.map((x) => x.assets.map((a) => a.tokenId));
+  const flatTokenList = tokenList.flat();
```
Comment (deadit) [inline src/utils.js]: ```const flatTokenList = outputs.flatMap(output  => outputs.assets.map(asset => asset.tokenId))```

File: src/index.js [hunk 3]
```diff
@@ -26,28 +23,20 @@ export const getAssetsFromBoxes = (boxes) => {
...
+      if (currAsset.tokenId in assetDict) {
+        assetDict[currAsset.tokenId] += currAsset.amount;
+      } else {
+        assetDict[currAsset.tokenId] = currAsset.amount;
+      }
+    }
+  }
 
-  return assets;
+  return Object.entries(assetDict).map((x) => ({ tokenId: x[0], amount: x[1] }));
```
[the remark this answers is not in the document]
Comment (Boaph5Du) [reply src/index.js]: done

File: src/index.js [hunk 4]
```diff
@@ -26,28 +23,20 @@ export const getAssetsFromBoxes = (boxes) => {
...
-  const allAssets = boxes
-    |> flatMap((box) => box.assets);
-
-  const allTokensIds = allAssets
-    |> map((asset) => asset.tokenId)
-    |> uniq;
+  const assetDict = {};
+  for (let i = 0; i < boxes.length; i += 1) {
+    for (let j = 0; j < boxes[i].assets.length; j += 1) {
+      const currAsset = boxes[i].assets[j];
```
[the remark this answers is not in the document]
Comment (Boaph5Du) [reply src/index.js]: done

File: src/utils.js [hunk 5]
```diff
@@ -40,7 +46,20 @@ export const inputBytes = (i) => {
...
+  const seenTokens = new Set();
+  const res = [];
+  for (let i = 0; i < flatTokenList.length; i += 1) {
+    const currId = flatTokenList[i];
+    if (!(currId in seenTokens)) {
+      res.push(currId);
+      seenTokens.add(currId);
+    }
+  }
+  return res;
```
[the remark this answers is not in the document]
Comment (Boaph5Du) [reply src/utils.js]: two tests committed

File: src/utils.js [hunk 6]
```diff
@@ -12,12 +12,18 @@ export const intToVlq = (num) => {
...
   res = Buffer.concat([res, intToVlq(out.assets.length)]);
+  for (let i = 0; i < out.assets.length; i += 1) {
+    const t = out.assets[i].tokenId;
+    const n = tokenIds.indexOf(t);
+    res = Buffer.concat([res, intToVlq(n)]);
+    res = Buffer.concat([res, intToVlq(out.assets[i].amount)]);
+  }
   const k = out.additionalRegisters.length;
   res = Buffer.concat([res, intToVlq(k)]);
   return res;
```
[the remark this answers is not in the document]
Comment (Boaph5Du) [reply src/utils.js]: done

File: src/utils.js (hunk 1 above)
Comment (Boaph5Du) [reply src/utils.js]: added raw byte dump in case tokenId not found (consistent with Ergo node implementation)

File: src/utils.js (hunk 2 above)
Comment (Boaph5Du) [reply src/utils.js]: done

**Pull request merged**